src/controllers/PermissionsController.php

```diff
@@ -137,11 +137,16 @@ public function actionAddRelation($name)
 		$model = new PermissionRelForm();
 		if ($model->load(Yii::$app->request->post())) {
 			$model->setPermission($perm);
-			if ($model->addRelations()) {
-				Yii::$app->session->setFlash('success', 'New relations added successfully.');
-			} else {
-				$errors = $model->getFirstErrors();
-				Yii::$app->session->setFlash('warning', $errors ? reset($errors) : 'Some error occured.');
+			//TODO make it no possible to choose loop created items
+			try {
+				if ($model->addRelations()) {
+					Yii::$app->session->setFlash('success', 'New relations added successfully.');
+				} else {
+					$errors = $model->getFirstErrors();
+					Yii::$app->session->setFlash('warning', $errors ? reset($errors) : 'Some error occured.');
+				}
+			} catch (\Exception $e) {
+				Yii::$app->session->setFlash('warning', 'Hierarchy error');
```

Comment:
Please write more accurate comment:

Relations can't be added because of hierarchy loop or impossible nesting.